docs/metrics.md

```diff
@@ -81,6 +81,9 @@ The current available memory of the VM containers based on the rss. Type: Gauge.
 ### kubevirt_vm_container_free_memory_bytes_based_on_working_set_bytes
 The current available memory of the VM containers based on the working set. Type: Gauge.
 
+### kubevirt_vm_created_total
+Amount of VMs created, broken down by namespace. Type: Counter.
+
```

Comment:
providers using KubeVirt (through insights) and users would benefit from this because it would ease the process of understanding the VM usage through time (with absolute values and increase/delta/etc functions to understand trends)

Reply:
The counter appears to be incremented whenever a VM is:

1.  Created for the first time 
2.  Deleted and recreated
3.  Reconciled the first time by a new instance of virt-controller

On a system with a lot of churn there will be a big drop when virt-controller is restarted.  Maybe that's not a big deal?  How is that handled by consumers?

Reply:
Prometheus handles the counter restarting at 0, when `virt-controller` is restarted.

The only issue here would be point 3: "Reconciled the first time by a new instance of virt-controller". But with:

```
// VM is nil or already processed
	if vm == nil || len(vm.Status.Conditions) != 0 {
		return
	}
```

that I added in `func NewVMCreated(vm *v1.VirtualMachine)`, since the VM would have the status conditions set, I would expect it not to happen, am I missing something?